src/wp-includes/media.php

```diff
@@ -1958,6 +1971,104 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 	return $image;
 }
 
+/**
+ * Adds optimization attributes to an `img` HTML tag.
+ *
+ * @since 6.3.0
+ *
+ * @param string $image   The HTML `img` tag where the attribute should be added.
+ * @param string $context Additional context to pass to the filters.
+ *
+ * @return string Converted `img` tag with optimization attributes added.
+ */
+function wp_img_tag_add_loading_optimization_attrs( $image, $context ) {
+	$width              = preg_match( '/ width="([0-9]+)"/', $image, $match_width ) ? (int) $match_width[1] : '';
```

Comment:
Took this from an existing `wp_image_add_srcset_and_sizes( $image, $image_meta, $attachment_id )` and it is found to work as intended.

Reply:
We should consider using the `WP_HTML_Tag_Processor` class here instead of regex.

```php
$el = new WP_HTML_Tag_Processor( $image );
$el->next_tag();

$width = $el->get_attribute('width');
$height = $el->get_attribute('height');
```

Reply:
Thank you @joemcgill. I have made this change.